### PR TITLE
[#103870] Make training requests an optional feature

### PR DESCRIPTION
**Pull request by drench**
Default is on, but when off it should act as it did before https://github.com/tablexi/nucore-open/pull/298 and display a message to the user about contacting the facility for approval.

### CODE REVIEW

File: config/locales/en.models.yml [hunk 1]
```diff
@@ -55,6 +55,11 @@ en:
           requires_approval: The %{product} requires approval to reserve. Would you like to be contacted by the core for access or training?
         instrument:
           not_available: The %{instrument} instrument is currently unavailable for reservation online.
+          requires_approval_html: |
+            The %{instrument} instrument requires approval to reserve;
+            please contact the facility for further information:<br/><br/>
+            %{facility}<br/>
+            <a href=\"mailto:%{email}\">%{email}</a>
```
Comment (drench) [inline config/locales/en.models.yml]: This is a reformatted revert of 110cf0ed0cf5377276c5965b1824fbe49203063c

File: app/controllers/products_common_controller.rb [hunk 2]
```diff
@@ -77,11 +77,16 @@ def show
 
     # is the user approved?
     if add_to_cart && !@product.can_be_used_by?(acting_user) && !session_user_can_override_restrictions?(@product)
-      if TrainingRequest.submitted?(session_user, @product)
-        flash[:notice] = t_model_error(Product, "already_requested_access", product: @product)
-        return redirect_to facility_path(current_facility)
+      if SettingsHelper.feature_on?(:training_requests)
```
Comment (jhanggi) [inline app/controllers/products_common_controller.rb]: Is there a way we can share this code with the instruments controller? It looks like it's the same other than the `@instrument` vs `@product`

File: spec/controllers/bundles_controller_spec.rb [hunk 3]
```diff
@@ -93,9 +93,29 @@
...
-          assert_redirected_to(new_facility_product_training_request_path(facility, bundle))
+        around(:each) do |example|
+          saved_setting = SettingsHelper.feature_on?(:training_requests)
+          SettingsHelper.enable_feature(:training_requests, training_requests)
+          example.run
+          SettingsHelper.enable_feature(:training_requests, saved_setting)
+        end
+
+        context "if the training request feature is enabled" do
+          let(:training_requests) { true }
```
Comment (jhanggi) [inline spec/controllers/bundles_controller_spec.rb]: Can we make this variable something more like `training_requests_enabled` to make it a little more readable in the `around`?

File: spec/controllers/items_controller_spec.rb [hunk 4]
```diff
@@ -89,9 +89,29 @@
           do_request
         end
 
-        it "gives the user the option to submit a request for approval" do
-          expect(assigns[:add_to_cart]).to be_blank
-          assert_redirected_to(new_facility_product_training_request_path(facility, item))
+        around(:each) do |example|
```
Comment (jhanggi) [inline spec/controllers/items_controller_spec.rb]: Is there a way we can come up with a better technique than duplicating this everywhere? We have a `include_context "feature enabled", :create_users`, in one place, but I'm not sure that's the best option. Is there something we could do with metadata?

```
context "with training requests", feature_setting: { training_requests: true } do
```
or something similar; that was just the first thing that popped in my mind.

**Comment by jhanggi** [comment]
Overall, looks pretty good; just a few minor thoughts about how to clean things up a bit.

**Comment by jhanggi** [comment]
:+1:

File: app/controllers/products_common_controller.rb (hunk 2 above)
Comment (drench) [inline app/controllers/products_common_controller.rb]: I'm working with @danhodos and @yoon on refactoring this. It's significant enough to deserve its own PR.
